lib/map.js

```diff
@@ -186,7 +186,11 @@ Mapper.prototype.push = function (data) {
   if (! this.isMulti() || data === null) {
     return push.call(this, data)
   } else {
-    return data.every(push.bind(this))
+    var needMoreData = false
+    for (var i = 0; i < data.length; ++i) {
+      needMoreData = push.call(this, data[i])
```

Comment:
It looks fine to me as is. You really only care about the value of the last `push()`.

`needsMore && push()` is basically the bug that existed before, and `needsMore || push()` will stop pushing when `needsMore` is true, which I think is the opposite of what you want.